### ISSUE TITLE
Level (⍚) does not behave as expected when second argument is has 1 less dimension than the first

### ISSUE THREAD

**Issue by dexterdy** [uiua-lang/uiua#37]
Maybe I'm interpreting the documentation wrong, but it seems to me like

```
⍚[1 1]⊂ [.↯3_3⇡9] [10_11_12]
```
should output
```
╭─                
╷ 0 1 2 10 11 12  
╷ 3 4 5 10 11 12  
  6 7 8 10 11 12  
                  
  0 1 2 10 11 12  
  3 4 5 10 11 12  
  6 7 8 10 11 12  
                 ╯
```
But it instead outputs
```
╭─                
╷ 0 1 2 10 11 12  
╷ 3 4 5 10 11 12  
  6 7 8 10 11 12  
                 ╯
```
Am I wrong? Could someone help explain this to me?

**Comment by kaikalii** [comment]
This is a bug, not because it is giving you the wrong output, but because it is not erroring.
You are doing a `⍚[1 1]` on arrays with shapes `[2 3 3]` and `[1 3]`.
At the level that you are drilling down to, the shape prefixes are `[2 3]` and `[1]`, which are not compatible.
With a `[3]` second array, you should get the expected result.
```
⍚[1 1]⊂ [.↯3_3⇡9] [10 11 12]
```

**Comment by kaikalii** [comment]
Fixed in 3a4b02c2d51510c

**Comment by dexterdy** [comment]
thank you!
